tests/integration/gcp.py

```diff
@@ -440,27 +438,25 @@ def _get_image(self, project, image_name):
         self.logger.debug(f"Looking for {image_name=} in {project=}")
         try:
             response = (
-                self._compute.images().get(project=project, image=image_name,).execute()
+                self._compute_images.get(project=project, image=image_name,)
             )
-            image = response.get("selfLink")
+            image = response.self_link
             return image
         except:
             self.logger.debug(f"Failed to find image in {project=}")
             return None
 
     def _wait_until_reachable(self, hostname):
-        self.logger.info(f"Waiting for {hostname} to respond to ping ...")
-        while True:
-            response = os.system("ping -c 1 " + hostname)
+        self.logger.info(f"Waiting for {hostname} to respond...")
+        i=0
+        while i<20:
+            response = os.system("timeout 1 bash -c \"</dev/tcp/" + hostname + "/22\"")
             if response == 0:
-                self.logger.info(f"Instance {hostname} is reachable, waiting another 20 seconds...")
-                time.sleep(20)
+                self.logger.info(f"Instance {hostname} is reachable...")
                 return
-            time.sleep(1)
-
-    def _get_resource_name(self, url):
-        """Get resource name from GCP url"""
-        return urlparse(url).path.split("/")[-1]
+            self.logger.info(f"Waiting for {hostname} to respond...")
+            time.sleep(2)
+            i += 1
```

Comment:
We should probably raise some sort of exception, shouldn't we? If we waited 20 seconds for the SSH server to get available and it's still not there, we should raise something like this:
```
raise Exception("Remote host unreachable")
```

In the upper function, we could then catch it with a `try ... except` block to handle this error properly and maybe cancel the whole execution.